### ISSUE TITLE
HardwareKeyboard Event Handler cannot be trigger until a soft keyboard is opened

### ISSUE THREAD

**Issue by live9080** [flutter/flutter#124101]
## Issue Summary

I am trying to listen to Bluetooth keyboard events in the camera page to trigger some business logic. 

I have tried using `HardwareKeyboard.instance.addHandler` in the `initState` lifecycle but I am unable to obtain the key event when I enter the page. I can only obtain the key event after the soft keyboard is opened, even if the soft keyboard is open on a different navigator page. 

Once the soft keyboard is opened, the HardwareKeyboard.instance.addHandler in other pages can be triggered directly without any further focus.

## Steps to Reproduce
1. Enter the camera page.
2. Try to obtain Bluetooth keyboard events using HardwareKeyboard.instance.addHandler in the initState lifecycle.

**Expected results:** 
I expect to obtain the Bluetooth keyboard event once I enter the app.

**Actual results:** 
I cannot obtain the Bluetooth keyboard event when I enter the page, and I can only obtain it after the soft keyboard is opened. 

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Flutter Demo',
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});
  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;
  HardwareKeyboard? hardwareKeyboard;

  @override
  void initState() {
    super.initState();

    hardwareKeyboard = HardwareKeyboard.instance;
    hardwareKeyboard?.addHandler(keyBoardHandler);
  }

  @override
  void dispose() {
    super.dispose();

    hardwareKeyboard?.removeHandler(keyBoardHandler);
  }

  /// The Audio Volume Up/Down Button could only be trigger after the TextField is focus(The soft keyboard is opened);
  bool keyBoardHandler(KeyEvent event) {
    print(event.logicalKey.keyLabel);
    setState(() {
      _counter++;
    });
    return false;
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text('$_counter'),
            TextField(),
          ],
        ),
      ),
    );
  }
}
```

</details>

```
flutter doctor -v
[✓] Flutter (Channel stable, 3.7.9, on macOS 13.2.1 22D68 darwin-arm64, locale zh-Hans-CN)
    • Flutter version 3.7.9 on channel stable at /Users/marvin/fvm/versions/stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 62bd79521d (5 days ago), 2023-03-30 10:59:36 -0700
    • Engine revision ec975089ac
    • Dart version 2.19.6
    • DevTools version 2.20.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/marvin/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.
```
</details>

**Comment by iapicca** [comment]
@live9080
are we sure that the bt keeb is not being seen
and not just the typing being visible only after focusing on a text field
(soft keeb opening being just a side effect of this)

maybe code sample + video could help

**Comment by darshankawar** [comment]
@live9080 
Does the same behavior occur using a physical device ? I am wondering if this is emulator specific issue or not.
See if [this](https://github.com/flutter/flutter/issues/71144) issue is related to your use case.

**Comment by live9080** [comment]
@iapicca 
Yes, I tested with a full-size keyboard with pixel 6. The other keys expecting "F1-F12" and "Media Keys" work well.

Could you try press  "F1-F12" or "Media Keys" such "volume up/down" first once you enter the app? Thanks
Once other keys such as "A-Z" was pressed.  The "F1-F12" or "Media Keys"  will work well too.

**Comment by darshankawar** [comment]
Thanks for the update. I am going ahead and closing this as a duplicate of https://github.com/flutter/flutter/issues/71144. You may share your findings in that issue for better tracking.


Please do leave a thumbs up on the original author's post so we can track and prioritize the issue base on severity and demand. Thanks!
